File: FragmentTp/app/src/main/java/com/example/fragmenttp/MainActivity.java
package com.example.fragmenttp;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentResultListener;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button frg1 = findViewById(R.id.btnFrgOne);
        Button frg2 = findViewById(R.id.btnFrgTwo);

        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentBoss,new FirstFragment()).commit();


        SecondFragment fragment2 = new SecondFragment();
        FirstFragment fragment1 = new FirstFragment();

        getSupportFragmentManager().setFragmentResultListener("msg", this, new FragmentResultListener() {
        @Override
        public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle result) {
            fragment2.setArguments(result);
        }
    });





        frg1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getSupportFragmentManager().beginTransaction().replace(R.id.fragmentBoss,fragment1).commit();
            }
        });

        frg2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getSupportFragmentManager().beginTransaction().replace(R.id.fragmentBoss,fragment2).commit();
            }
        });



    }
}

File: SWIPABLE_VIEWS/app/src/main/java/com/lahbouch/swipable_views/MainActivity.kt
package com.lahbouch.swipable_views

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.viewpager2.widget.ViewPager2

class MainActivity : AppCompatActivity() {
    lateinit var vpager : ViewPager2
    lateinit var adapter : ImgAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val data = mutableListOf<Int>(
            R.drawable.p1,
            R.drawable.p,
            R.drawable.p2,
            R.drawable.p4,
            R.drawable.p5
        )
        adapter = ImgAdapter(data)
        vpager = findViewById(R.id.viewPager)
        vpager.adapter = adapter

        vpager.orientation = ViewPager2.ORIENTATION_VERTICAL

        vpager.beginFakeDrag()
        vpager.fakeDragBy(-10f)
        vpager.endFakeDrag()

    }
}